packages/tailwindcss/src/variants.test.ts

```diff
@@ -2155,6 +2162,14 @@ test('data', async () => {
       display: flex;
     }
 
+    .data-\\[potato_\\=_\\"salad\\"\\]\\:flex[data-potato="salad"] {
+      display: flex;
+    }
+
+    .data-\\[potato_\\^\\=_\\"salad\\"\\]\\:flex[data-potato^="salad"] {
```

Comment:
Is it lightningcss that did remove the spaces again here or did we? I feel like if the argument is "we want to support arbitrary CSS because CSS would also allow that", we should also make sure we output the spaces again in the resulting CSS. Conceptually we just pass-through even though we do auto-quote where we have to.

Reply:
Yep Lightning CSS is cleaning that up for us: https://lightningcss.dev/playground/index.html#%7B%22minify%22%3Afalse%2C%22customMedia%22%3Atrue%2C%22cssModules%22%3Afalse%2C%22analyzeDependencies%22%3Afalse%2C%22targets%22%3A%7B%22chrome%22%3A6225920%7D%2C%22include%22%3A0%2C%22exclude%22%3A0%2C%22source%22%3A%22.data-%5C%5C%5Bpotato_%5C%5C%5E%5C%5C%3D_%5C%5C%5C%22salad%5C%5C%5C%22%5C%5C%5D%5C%5C%3Aflex%5Bdata-potato%20%5E%3D%20%5C%22salad%5C%22%5D%20%7B%5Cn%20%20color%3A%20red%3B%5Cn%7D%22%2C%22visitorEnabled%22%3Afalse%2C%22visitor%22%3A%22%7B%5Cn%20%20Color(color)%20%7B%5Cn%20%20%20%20if%20(color.type%20%3D%3D%3D%20'rgb')%20%7B%5Cn%20%20%20%20%20%20color.g%20%3D%200%3B%5Cn%20%20%20%20%20%20return%20color%3B%5Cn%20%20%20%20%7D%5Cn%20%20%7D%5Cn%7D%22%2C%22unusedSymbols%22%3A%5B%5D%2C%22version%22%3A%22local%22%7D


I don't think re-emitting the _exact_ space is necessary (even though we do). This is what we generate:
```css
.data-\\[potato_\\^\\=_\\"salad\\"\\]\\:flex {
  &[data-potato ^= "salad"] {
    display: flex;
  }
}
```

Only thing I care about is that if you pass in valid CSS that valid CSS comes out and that we didn't generate invalid CSS along the way.